src/openrct2/scenario/ScenarioRepository.cpp

```diff
@@ -272,6 +275,36 @@ class ScenarioRepository final : public IScenarioRepository
         delete scanner;
     }
 
+    void ScanMegaPark(std::string &dataDir, std::string &scenarioDir)
+    {
+        //Convert mp.dat from RCT1 Data directory into SC21.sc4 (Mega Park)
+        utf8 mpdatPath[MAX_PATH];
+        utf8 sc21Path[MAX_PATH];
+
+        String::Set(mpdatPath, sizeof(mpdatPath), dataDir.c_str());
+        Path::Append(mpdatPath, sizeof(mpdatPath), "mp.dat");
+
+        String::Set(sc21Path, sizeof(sc21Path), scenarioDir.c_str());
+        platform_ensure_directory_exists(sc21Path); //Ensure the scenario directory exists first
+        Path::Append(sc21Path, sizeof(sc21Path), "SC21.SC4");
+
+        if (platform_file_exists(mpdatPath) && !(platform_file_exists(sc21Path)))
+        {
+            uint8 byte;
+            size_t length;
+            auto mpdat = (uint8 *)(File::ReadAllBytes(mpdatPath, &length));
+            auto outFS = FileStream(sc21Path, FILE_MODE_WRITE);
+
+            for (uint32 i = 0; i < (uint32)length; i++)
+            {
+                //Rotate each byte left by 4 bits to convert
+                byte = *(mpdat + i);
+                byte = rol8(byte, 4); 
+                outFS.Write(&byte, sizeof(byte));
```

Comment:
This would be very inefficient, it would be better to read all the data into memory, `rol8` it and only then write it all.